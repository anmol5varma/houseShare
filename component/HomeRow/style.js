export default {
    container: {
        flexDirection: 'row',
    },
    name: {
        fontSize: 20,
        textAlign: 'center',
        color: '#111',
    },
    address: {
        fontSize: 16,
        textAlign: 'center',
        color: '#111',
    },
    thumbnail:{
        width: 100,
        height: 100,
        borderWidth: 0.5,
        borderColor: '#d6d7da',
    },
    details:{
        flex: 1,
        flexDirection: 'column',
        justifyContent: 'center',
    },
}