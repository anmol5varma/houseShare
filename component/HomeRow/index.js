/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * @format
 * @flow
 * @lint-ignore-every XPLATJSCOPYRIGHT1
 */

import React, { Component } from 'react';
import { Image, StyleSheet, Text, View } from 'react-native';
import style from './style'

const styles = StyleSheet.create(style);
export default class HomeRow extends Component {
    render() {
        return (
            <View style={styles.container}>
                <View><Image style={styles.thumbnail} source={{ uri: this.props.src }} /></View>
                <View style={styles.details}>
                    <Text style={styles.name}>{this.props.name}</Text>
                    <Text style={styles.address}>{this.props.address}</Text>
                </View>
            </View>
        );
    }
}
