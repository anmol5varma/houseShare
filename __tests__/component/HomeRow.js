/**
 * @format
 * @lint-ignore-every XPLATJSCOPYRIGHT1
 */

import 'react-native';
import React from 'react';
import HomeRow from '../../component/HomeRow';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(<HomeRow name='Test name' address='Test address' src='sample image' />).toJSON();
  expect(tree).toMatchSnapshot();
});
