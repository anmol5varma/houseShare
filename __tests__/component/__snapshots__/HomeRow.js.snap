// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
    }
  }
>
  <View>
    <Image
      source={
        Object {
          "uri": "sample image",
        }
      }
      style={
        Object {
          "height": 100,
          "width": 100,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#111",
          "fontSize": 20,
          "textAlign": "center",
        }
      }
    >
      Test name
    </Text>
    <Text
      style={
        Object {
          "color": "#111",
          "fontSize": 16,
          "textAlign": "center",
        }
      }
    >
      Test address
    </Text>
  </View>
</View>
`;
